src/jdk.compiler/share/classes/com/sun/tools/javac/parser/JavaTokenizer.java

```diff
@@ -1273,76 +1261,104 @@ public CommentStyle getStyle() {
          * @return true if comment contains @deprecated.
          */
         public boolean isDeprecated() {
-            if (!scanned && cs == CommentStyle.JAVADOC) {
+            if (!scanned) {
                 scanDocComment();
             }
-
             return deprecatedFlag;
         }
 
         /**
-         * Scan JAVADOC comment for details.
+         * Detect the deprecated tag.
+         *
+         * @param line line reader
+         *
+         * @return true if deprecated tag is present.
          */
-        protected void scanDocComment() {
-            try {
-                boolean deprecatedPrefix = false;
-                accept("/**");
+        protected boolean hasDeprecated(UnicodeReader line) {
+            return line.accept("@deprecated") &&
+                    (line.isWhitespace() || line.is('*') || line.isEOLN());
+        }
```

Comment:
I note for the record that this is inadequate, given that we now skip over such content in inline tags. This is an unexpected side-effect of that change.

I am not suggesting we do anything about it at this time, but it is worth noting.

Reply:
This code is basically maintain the status quo - in a more efficient manner.